thirdparty/patches/arrow-16.1.0-use-zstd-1.5.0.patch

```diff
@@ -0,0 +1,15 @@
+diff --git a/cpp/thirdparty/versions.txt b/cpp/thirdparty/versions.txt
+index 67275ffea..983744990 100644
+--- a/cpp/thirdparty/versions.txt
++++ b/cpp/thirdparty/versions.txt
+@@ -117,8 +117,8 @@ ARROW_XSIMD_BUILD_VERSION=13.0.0
+ ARROW_XSIMD_BUILD_SHA256_CHECKSUM=8bdbbad0c3e7afa38d88d0d484d70a1671a1d8aefff03f4223ab2eb6a41110a3
+ ARROW_ZLIB_BUILD_VERSION=1.3.1
+ ARROW_ZLIB_BUILD_SHA256_CHECKSUM=9a93b2b7dfdac77ceba5a558a580e74667dd6fede4585b91eefb60f03b72df23
+-ARROW_ZSTD_BUILD_VERSION=1.5.6
+-ARROW_ZSTD_BUILD_SHA256_CHECKSUM=8c29e06cf42aacc1eafc4077ae2ec6c6fcb96a626157e0593d5e82a34fd403c1
++ARROW_ZSTD_BUILD_VERSION=1.5.0
```

Comment:
![image](https://github.com/StarRocks/starrocks/assets/771299/962664fa-1e4a-4d91-a499-8b9efac13b1e)


is it possible to disable this auto zstd download and building, guide arrow to use the zstd version our thirdparty provided?

Reply:
<img width="686" alt="image" src="https://github.com/StarRocks/starrocks/assets/16740944/1e86f65b-8ab8-4c0e-8cf5-164524a53413">
Looks like our building script requires this downloading step.

Reply:
besides we have provided 
```
export ARROW_ZSTD_URL=${TP_SOURCE_DIR}/${ZSTD_NAME}
```
which direct the build process to read from local file.